### PR TITLE
LAWS-151 Missouri: St Louis Earnings Tax AND Employer Payroll Expense Tax

### PR DESCRIPTION
**Pull request by Aidancs**
refs: https://spurwork.atlassian.net/browse/LAWS-151

### CODE REVIEW

File: src/migrations/2020_03_11_000003_add_st_louis_taxes.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            'home_governmental_unit_area_id' => $st_louis_gua_id,
+            'work_governmental_unit_area_id' => $st_louis_gua_id,
+            'based' => TaxArea::BASED_ON_WORK_LOCATION,
+        ]]);
+
+        DB::table($this->tax_areas)->insert([[
+            'tax_id' => $st_louis_employer_tax,
+            'work_governmental_unit_area_id' => $st_louis_gua_id,
+            'based' => TaxArea::BASED_ON_WORK_LOCATION,
+        ]]);
```
Comment (stevespur) [inline src/migrations/2020_03_11_000003_add_st_louis_taxes.php]: `A St Louis Earnings tax must be withheld from all wages of residents of the city, and from wages earned within the city from non-residents.`

shouldn't these be based on either?

File: src/Countries/US/Missouri/StLouis/V20200101/StLouis.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Appleton\Taxes\Countries\US\Missouri\StLouis\V20200101;
+
+use Appleton\Taxes\Countries\US\Missouri\StLouis\StLouis as BaseStLouis;
+
+use Illuminate\Database\Eloquent\Collection;
+
+class StLouis extends BaseStLouis
+{
+    const TAX_RATE = 0.001;
```
Comment (stevespur) [inline src/Countries/US/Missouri/StLouis/V20200101/StLouis.php]: isn't this .1%

File: src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Appleton\Taxes\Countries\US\Missouri\StLouisEmployer\V20200101;
+
+use Appleton\Taxes\Countries\US\Missouri\StLouisEmployer\StLouisEmployer as BaseStLouisEmployer;
+use Illuminate\Database\Eloquent\Collection;
+
+class StLouisEmployer extends BaseStLouisEmployer
+{
+    const TAX_RATE = 0.0005;
```
Comment (stevespur) [inline src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php]: isn't this .05%?

File: src/migrations/2020_03_11_000003_add_st_louis_taxes.php (hunk 1 above)
Comment (Aidancs) [reply src/migrations/2020_03_11_000003_add_st_louis_taxes.php]: Ac 1 is for the earnings tax and yes it should be weither

Ac 4 is for the employers tax and it is "4. EMPLOYER liability for "Payroll Expense Tax" should be created on all wages earned in St. Louis. " so I think the second one is right

File: src/Countries/US/Missouri/StLouis/V20200101/StLouis.php (hunk 2 above)
Comment (Aidancs) [reply src/Countries/US/Missouri/StLouis/V20200101/StLouis.php]: oops good catch

File: src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php (hunk 3 above)
Comment (Aidancs) [reply src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php]: Oops good catch, apparently I need to go back to middle school

File: src/migrations/2020_03_11_000003_add_st_louis_taxes.php (hunk 1 above)
Comment (stevespur) [reply src/migrations/2020_03_11_000003_add_st_louis_taxes.php]: ya, I didn't pay enough attention :p

File: src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php (hunk 3 above)
Comment (stevespur) [reply src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php]: I had to think about it and double check myself

File: src/migrations/2020_03_11_000003_add_st_louis_taxes.php (hunk 1 above)
Comment (Aidancs) [reply src/migrations/2020_03_11_000003_add_st_louis_taxes.php]: All good, thanks you for reviewing and checking, I had to double check to make sure.

File: src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php (hunk 3 above)
Comment (Aidancs) [reply src/Countries/US/Missouri/StLouisEmployer/V20200101/StLouisEmployer.php]: I literally googled it to make sure. I was looking at them thinking "No, they're right"

**Pull request merged**